### ISSUE TITLE
[Feature Request] Add robot9000 mode option to remove repeated spam messages to improve performance further

### ISSUE THREAD

**Issue by Zallist** [LiveTL/HyperChat#10]
One of the things that makes the chat most unreadable on YouTube is the repeated spam of message after message saying the same thing in a short period. This also adds to the performance overhead. A simple check for if a message has been repeated during the current stream (or a set number of minutes) goes a long way to cleaning down the spam messages.

I've made a TamperMonkey UserScript which strips out duplicate messages via overriding the `window.fetch` in a sliding window that could be used as a base here:
https://greasyfork.org/en/scripts/427292-cleanup-youtube-live-chat-for-spam-r9k-mode/code

Simply put, the algorithm works as:
- Make a message cache
- Check the message cache for a stripped version of the message (with emojis, without emojis) within a certain timeframe of the message time, getting the count
- If the count is over a threshold, remove the message
- Log the stripped version of the message & the time it would have appeared

Relevant xkcd (but not really): https://blog.xkcd.com/2008/01/14/robot9000-and-xkcd-signal-attacking-noise-in-chat/

**Comment by Candygoblen123** [comment]
This seems a little out of scope for Hyperchat. If hyperchat breaks your script, then you can submit something to make sure it works. however, i don't think we want to include this in the extension

**Comment by Candygoblen123** [comment]
to clarify: if hyperchat breaks your tamper monkey script, let us know. but we aren't going to add this feature to the extension itself
